Package Request: lrzsz zssh

I am unix Administrator ,I need legacy file transfer via ssh for low speed connection. so I try to build these two packages for it.

globbing.o: In function `glob_args':
/data/data/com.termux/files/home/zssh-1.5c/globbing.c:34: undefined reference to `glob'
/data/data/com.termux/files/home/zssh-1.5c/globbing.c:42: undefined reference to `globfree'
clang-3.9: error: linker command failed with exit code 1 (use -v to see invocation)

roid-glob
make: [Makefile:74: ztelnet] Error 1 (ignored)
ln zssh ztelnet
ln: failed to create hard link 'ztelnet' => 'zssh': Permission denied
make: *** [Makefile:75: ztelnet] Error 1                                                                            $ ./zssh
Segmentation fault (core dumped)
$ ./zssh --hel[
Segmentation fault (core dumped)
i had not changed the source code.but it fail.

--- a/packages/zssh/Makefile.in.patch
+++ b/packages/zssh/Makefile.in.patch
@@ -1,0 +1,21 @@
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -71,13 +71,13 @@
+ 
+ ztelnet : $(NAME)
+ 	@-rm $@ > /dev/null 2>&1
+-	ln $(NAME) $@
++	ln -s $(NAME) $@
+ 
+ install :
+-	strip $(NAME)
+-	$(INSTALL) -m 0711 zssh @bindir@
+-	ln -f @bindir@/zssh  @bindir@/ztelnet
+-	$(INSTALL) -m 0644 $(MANS) @mandir@/man1
++	$(STRIP) $(NAME)
++	$(INSTALL) -Dm 0711 -t @bindir@ zssh
++	ln -fs @bindir@/zssh  @bindir@/ztelnet
++	$(INSTALL) -Dm 0644 -t @mandir@/man1 $(MANS) 
+ 
+ uninstall :
+ 	-cd @bindir@      && rm $(PRGS)

--- a/packages/zssh/build.sh
+++ b/packages/zssh/build.sh
@@ -1,0 +1,14 @@
+TERMUX_PKG_HOMEPAGE=http://zssh.sourceforge.net/
+TERMUX_PKG_DESCRIPTION="A program for interactively transferring files to a remote machine while using the secure shell (ssh)"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=1.5c
+TERMUX_PKG_SRCURL=https://downloads.sourceforge.net/zssh/zssh-${TERMUX_PKG_VERSION}.tgz
+TERMUX_PKG_SHA256=a2e840f82590690d27ea1ea1141af509ee34681fede897e58ae8d354701ce71b
+TERMUX_PKG_DEPENDS="libandroid-glob, lrzsz, openssh, readline"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_pre_configure() {
+	echo "ac_cv_func_getpgrp_void=yes" >> config.cache
+	LDFLAGS+=" -landroid-glob"
+}

--- a/packages/zssh/escape.c.patch
+++ b/packages/zssh/escape.c.patch
@@ -1,0 +1,58 @@
+--- a/escape.c
++++ b/escape.c
+@@ -10,11 +10,8 @@
+ 
+ #include "zssh.h"
+ 
+-int		pc_escape_double_chr(str,i,chr,c2)
+-char		*str;
+-int		*i;
+-char		chr;
+-char		c2;
++int
++pc_escape_double_chr(char *str, int *i, char chr, char c2)
+ {
+    int		j;
+    
+@@ -32,11 +29,8 @@
+    return (1);
+ }
+ 
+-int		pc_escape_par(str,i,c1,c2)
+-char		*str;
+-int		*i;
+-char		c1;
+-char		c2;
++int
++pc_escape_par(char *str, int *i, char c1, char c2)
+ {
+    int		n;
+    int		j;
+@@ -65,11 +59,8 @@
+ }
+ 
+ 
+-int		pc_escape_backslash(str,i,c1,c2)
+-char		*str;
+-int		*i;
+-char		c1;
+-char		c2;
++int
++pc_escape_backslash(char *str, int *i, char c1, char c2)
+ {
+    if (str[*i] != '\\')
+       return (0);
+@@ -82,11 +73,8 @@
+ 
+ 
+ 
+-int		pc_escape_dollar_par(str,i,c1,c2)
+-char		*str;
+-int		*i;
+-char		c1;
+-char		c2;
++int
++pc_escape_dollar_par(char *str, int *i, char c1, char c2)
+ {
+    int		n;
+    int		j;

--- a/packages/zssh/misc.c.patch
+++ b/packages/zssh/misc.c.patch
@@ -1,0 +1,24 @@
+--- a/misc.c
++++ b/misc.c
+@@ -11,8 +11,8 @@
+ #include "zssh.h"
+ 
+ 
+-char            *chr2str(chr)
+-char            chr;
++char *
++chr2str(char chr)
+ {
+    char         *pt;
+ 
+@@ -32,8 +32,8 @@
+ };
+ 
+ 
+-int     mi_is_whitespace(chr)
+-char    chr;
++int
++mi_is_whitespace(char chr)
+ {
+    int  j;
+    

--- a/packages/zssh/openpty.c.patch
+++ b/packages/zssh/openpty.c.patch
@@ -1,0 +1,13 @@
+--- a/openpty.c
++++ b/openpty.c
+@@ -10,6 +10,10 @@
+ 
+ #include "zssh.h"
+ 
++#ifdef __ANDROID__
++#include <pty.h>
++#endif
++
+ #define GL_SLAVENAMELEN	50
+ static char	gl_slavename[GL_SLAVENAMELEN + 1] = {0};
+ 

--- a/packages/zssh/quote_removal.c.patch
+++ b/packages/zssh/quote_removal.c.patch
@@ -1,0 +1,15 @@
+--- a/quote_removal.c
++++ b/quote_removal.c
+@@ -23,10 +23,8 @@
+    return (1);
+ }
+ 
+-int		pc_remove_double_chr(str,i,chr)
+-char		*str;
+-int		*i;
+-char		chr;
++int
++pc_remove_double_chr(char *str, int *i, char chr)
+ {
+    str_shift(str,*i,1);
+    while (str[*i] && str[*i] != chr)

--- a/packages/zssh/zmodem_act.c.patch
+++ b/packages/zssh/zmodem_act.c.patch
@@ -1,0 +1,23 @@
+--- a/zmodem_act.c
++++ b/zmodem_act.c
+@@ -113,6 +113,9 @@
+ char	**av;
+ int	master;
+ {
++   char cmd[10] = "l";
++   strncat(cmd, av[0], sizeof(cmd)-1);
++   if (!sfork(&gl_child_rz))
+    tcsetattr(gl_slave, TCSAFLUSH, &gl_rtt);
+    if (!sfork(&gl_child_rz))
+    {
+@@ -120,8 +123,8 @@
+       signal(SIGWINCH,SIG_DFL);
+       dup2(master,0);
+       dup2(master,1);
+-      execvp(av[0],av);
+-      error("error: execvp %s\n",av[0]);
++      execvp(cmd,av);
++      error("error: execvp %s\n",cmd);
+       exit (1);
+    }
+ #ifdef DEBUG

